pkg/util/kubernetes/apiserver/leaderelection/leaderelection.go

```diff
@@ -61,8 +62,11 @@ type LeaderEngine struct {
 	coordClient         coordinationv1.CoordinationV1Interface
 	ServiceName         string
 	leaderIdentityMutex sync.RWMutex
-	leaderElector       *leaderelection.LeaderElector
-	lockType            string
+	// leaderIPMutex is used to avoid concurrent calls to GetLeaderIP
+	// that would result in multiple calls to the API server
+	leaderIPMutex sync.Mutex
```

Comment:
The cache and the mutex were not required before, why are they needed now? Is it because we expect a substantial increase in the number of request and this is needed for performance reasons?

Reply:
I think that it might not be necessary, the number of requests should be greatly reduced thanks to the cache.